subsys/net/lib/lwm2m/lwm2m_obj_connmon.c

```diff
@@ -222,4 +222,4 @@ static int lwm2m_connmon_init(void)
 	return ret;
 }
 
-SYS_INIT(lwm2m_connmon_init, APPLICATION, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+LWM2M_CORE_INIT(lwm2m_connmon_init);
```

Comment:
Why core init? Obj 4 is optional.

Reply:
Core object.
It is defined in LwM2M Core specification: https://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#13-5-0-E5-LwM2M-Object-Connectivity-Monitoring

Not as a separate object published in registry.

It does not matter if its optional or not..